scripts/python/services/test_GithubService.py

```diff
@@ -1,60 +1,132 @@
+                                                                                              {"login": "john.smith"}]
+        response = GithubService("", ORGANISATION_NAME).get_outside_collaborators_login_names()
         self.assertEqual(["tom-smith", "john.smith"], response)
 
-    def test_get_outside_collaborators_login_names_calls_downstream_services(self):
-        self.mock_github.get_organization().get_outside_collaborators.return_value = []
-        self.github_service.get_outside_collaborators_login_names()
-        self.mock_github.get_organization.assert_has_calls(
+    def test_calls_downstream_services(self, mock_github):
+        mock_github.return_value.get_organization().get_outside_collaborators.return_value = []
+        github_service = GithubService("", ORGANISATION_NAME)
+        github_service.get_outside_collaborators_login_names()
+        github_service.client.get_organization.assert_has_calls(
             [call(), call(ORGANISATION_NAME), call().get_outside_collaborators()])
 
-    def test_get_outside_collaborators_login_names_returns_empty_list_when_collaborators_returns_none(self):
-        self.mock_github.get_organization().get_outside_collaborators.return_value = None
-        response = self.github_service.get_outside_collaborators_login_names()
+    def test_returns_empty_list_when_collaborators_returns_none(self, mock_github):
+        mock_github.return_value.get_organization().get_outside_collaborators.return_value = None
+        github_service = GithubService("", ORGANISATION_NAME)
+        response = github_service.get_outside_collaborators_login_names()
         self.assertEqual([], response)
 
-    def test_get_outside_collaborators_login_names_returns_exception_when_collaborators_returns_exception(self):
-        self.mock_github.get_organization(
+    def test_returns_exception_when_collaborators_returns_exception(self, mock_github):
+        mock_github.return_value.get_organization(
         ).get_outside_collaborators.side_effect = ConnectionError
+        github_service = GithubService("", ORGANISATION_NAME)
         self.assertRaises(
-            ConnectionError, self.github_service.get_outside_collaborators_login_names)
+            ConnectionError, github_service.get_outside_collaborators_login_names)
 
-    def test_get_outside_collaborators_login_names_returns_exception_when_organization_returns_exception(self):
-        self.mock_github.get_organization.side_effect = ConnectionError
+    def test_returns_exception_when_organization_returns_exception(self, mock_github):
+        mock_github.return_value.get_organization.side_effect = ConnectionError
+        github_service = GithubService("", ORGANISATION_NAME)
+        self.assertRaises(
+            ConnectionError, github_service.get_outside_collaborators_login_names)
+
+
+@patch("github.Github.__new__")
+class TestGithubServiceCloseExpiredIssues(unittest.TestCase):
```

Comment:
Refactored the test structure so each function gets its own class. The testing can be fairly verbose, so having the ability to fold away functions you don't want to see is nice.

I did try to nest classes, though I couldn't seem to get this to work.